packages/phone-number-privacy/common/src/utils/authentication.ts

```diff
@@ -16,6 +16,9 @@ import {
 } from '../interfaces'
 import { FULL_NODE_TIMEOUT_IN_MS, RETRY_COUNT, RETRY_DELAY_IN_MS } from './constants'
 
+import opentelemetry, { SpanStatusCode } from '@opentelemetry/api'
+const tracer = opentelemetry.trace.getTracer('signer-tracer')
```

Comment:
how do we set new tracers?

Reply:
The very basic way is just enclosing the code we want to instrument like this:

```node
        tracer.startActiveSpan('<NAME>', async (span) => {
          <YOUR_CODE_HERE>
          span.end() -> !! Don't forget to end the span
        })
```

- If the span is expected to have children (nested, like in the first function executed), use `tracer.startActiveSpan('<NAME>', async (parentSpan) => {` instead.
- If you are instrumenting a function that must return a value (this error `A function whose declared type is neither 'undefined', 'void', nor 'any' must return a value.`), add `return` at the beginning like this: `return tracer.startActiveSpan('Authentication - authenticateUser', async (span) => {`.

Apart from that, we can add events (like comments) with `span.addEvent(<STRING>)` or set the status like this:
```
      span.setStatus({
        code: SpanStatusCode.ERROR,
        message: 'No messageSignature or signer',
      })
```

Adding some refs:
- This page has almost all info for Node: https://opentelemetry.io/docs/instrumentation/js/manual/#traces

Reply:
Thanks for the docs.

If I understand correcly, the value in `trace.getTracer()` can be any uniquely identifyable string that describes the scope of the trace.

Reply:
Yes, that's what I get from the docs: https://opentelemetry.io/docs/instrumentation/js/manual/#acquiring-a-tracer

I'm linking it because I'm not really savvy on node or typescript, so don't wanna mislead you.